blog/2016-07-16-typescript-status-report.md

```diff
@@ -0,0 +1,111 @@
+---
+title: "Status Report: Ceylon TypeScript Loader GSoC project"
+author: Lucas Werkmeister
+layout: blog
+unique_id: blogpage
+tab: blog
+tags: [gsoc, typescript]
+---
+
+TL;DR: coming along, won’t be done in time for official GSoC deadline, I’ll just keep working.
+
+For this year’s Google Summer of Code, I am working on adding TypeScript support to the Ceylon JavaScript backend.
+The goal is to have a tool that, given a TypeScript module (one or more files), produces a Ceylon module for the JavaScript backend.
+The JS code of that module will just be the TypeScript compiler’s output (plus possibly some metamodel stuff),
+but the tool will also add model information that will allow you to use the TypeScript module from Ceylon like any other Ceylon module,
+without needing to use `dynamic` blocks or having to declare your own dynamic interfaces.
+
+I actually started work on this tool a few months before GSoC, in January this year.
+I wrote the first iteration of the program in TypeScript in order to be able to interact with the TypeScript compiler
+(which is also written in TypeScript).
+The goal of that first iteration is just to be able to load the TypeScript compiler itself,
+so that I can then use that module (`tsc`) from Ceylon
+and write the second (and probably final) iteration of the program in Ceylon.
+
+This first iteration of the program was mostly finished,
+at least good enough that I could start writing the second iteration,
+shortly after the GSoC work period officially started.
+I had hoped to start with the second iteration exactly at the start of GSoC,
+but while there had been good progress with my own simple test module,
+`tsc` held a couple of nasty surprises that I hadn’t anticipated and that I had to work around.
+It also turned out that I had made a couple of very bad decisions early on:
+
+- I chose to operate on the parsed syntax tree instead of the typechecker model.
+  How I ever expected this to work, I have no idea, but problems quickly became obvious:
+  
+  - When I see a type reference, like `Element`, I have to know if this refers to a type parameter,
+    a type from the same module, or perhaps one from another module.
+    The type I store in the in the model must be fully qualified.
```

Comment:
Looks good to me also.

On Mon, Jul 18, 2016 at 6:27 AM, Lucas Werkmeister <notifications@github.com
> wrote:

> In blog/2016-07-16-typescript-status-report.md
> <https://github.com/ceylon/ceylon-lang.org/pull/424#discussion_r71132297>:
>
> > +and write the second (and probably final) iteration of the program in Ceylon.
> > +
> > +This first iteration of the program was mostly finished,
> > +at least good enough that I could start writing the second iteration,
> > +shortly after the GSoC work period officially started.
> > +I had hoped to start with the second iteration exactly at the start of GSoC,
> > +but while there had been good progress with my own simple test module,
> > +`tsc` held a couple of nasty surprises that I hadn’t anticipated and that I had to work around.
> > +It also turned out that I had made a couple of very bad decisions early on:
> > +
> > +- I chose to operate on the parsed syntax tree instead of the typechecker model.
> > +  How I ever expected this to work, I have no idea, but problems quickly became obvious:
> > +
> > +  - When I see a type reference, like `Element`, I have to know if this refers to a type parameter,
> > +    a type from the same module, or perhaps one from another module.
> > +    The type I store in the in the model must be fully qualified.
>
> thanks, fixed :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ceylon/ceylon-lang.org/pull/424/files/9f67405d6a647e54e5e1e3deffc1ae0fdc10b243#r71132297>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABFD95hkyaRfll_cuPlto4bt1mnf6_Cks5qW2MzgaJpZM4JOEXt>
> .
>